deployment/bare_metal/start.sh

```diff
@@ -163,17 +192,52 @@ yarn install
 yarn build
 # TODO maybe handle this differently?
 export NODE_ENV=production
-pm2 delete gradido-dht-node
 if [ ! -z $FEDERATION_DHT_TOPIC ]; then
   pm2 start --name gradido-dht-node "yarn --cwd $PROJECT_ROOT/dht-node start" -l $GRADIDO_LOG_PATH/pm2.dht-node.$TODAY.log --log-date-format 'YYYY-MM-DD HH:mm:ss.SSS'
+  pm2 save
 else
-  echo "====================================================================="
-  echo "WARNING: FEDERATION_DHT_TOPIC not configured. DHT-Node not started..."  
-  echo "====================================================================="
+  echo "=====================================================================" >> $UPDATE_HTML
+  echo "WARNING: FEDERATION_DHT_TOPIC not configured. DHT-Node not started..."  >> $UPDATE_HTML
+  echo "=====================================================================" >> $UPDATE_HTML
 fi  
-pm2 save
+
+
+# Install & build federation
+echo 'Updating federation' >> $UPDATE_HTML
+cd $PROJECT_ROOT/federation
+# TODO maybe handle this differently?
+unset NODE_ENV
+yarn install
+yarn build
+# TODO maybe handle this differently?
+export NODE_ENV=production
+
+# set FEDERATION_PORT from FEDERATION_COMMUNITY_APIS
+IFS="," read -a API_ARRAY <<< $FEDERATION_COMMUNITY_APIS
+for api in "${API_ARRAY[@]}"
+do
+  export FEDERATION_API=$api
```

Comment:
This is where you override the config values and use command line parameters, which is the correct solution.

Reply:
no there a no commandline parameters, because in line 216 you see there is a reading from the property FEDERATION_COMMUNITY_APIS in the API_ARRAY, where input is a config value and no commandline parameter.

Reply:
Maybe commandline parameter is not the correct name - its `environment variables` which is basically the same.
The `FEDERATION_API` is fed into the Application and the program uses this value - thats why I say, please remove the values from the config .env file ( in this particular case - the `.env.template' file).

Reply:
Further read: https://www.npmjs.com/package/dotenv

![image](https://user-images.githubusercontent.com/1238238/223159789-9d671fcd-4b4a-47e1-a09a-7b553df2eefb.png)

That is exactly what happens here.

Reply:
That's the reason, why in the directory deployment these properties are not defined in .env.dist nor in the .env-file.
In the directory federation these properties are define in my local .env and so in the .env.dist and .env.dist.template too.

Isn't it the responsibility of the user/admin to take care about the correct definition of all properties in the .env file of each module?

So why should a property of federation-module should be removed, when it is used in this module.
The usage in the deployment/bare_metal differs from the usage of each single module on their own, right?

Reply:
done